apps/mobile/src/app/(home)/settings/notifications/index.tsx

```diff
@@ -1,22 +1,36 @@
+import { useRegisterNotifications } from '@/components/notifications/use-notifications';
 import { SettingsListItem } from '@/components/settings/settings-list-item';
+import { useSettings } from '@/store/settings/settings';
 import { t } from '@lingui/macro';
 
 import { Box, PlaceholderIcon } from '@leather.io/ui/native';
 
 export default function SettingsNotificationsScreen() {
+  const { changeNotificationsPreference, notificationsPreference } = useSettings();
+  const { registerNotifications, unregisterNotifications } = useRegisterNotifications();
   return (
     <Box bg="ink.background-primary" flex={1}>
       <Box flex={1} gap="3" paddingTop="5">
         <SettingsListItem
           title={t({
             id: 'notifications.push.cell_title',
-            message: 'Notification',
+            message: 'Transaction confirmations',
+          })}
+          caption={t({
+            id: 'notifications.push.cell_caption',
+            message: 'Receive updates when sending and receiving amounts',
           })}
           icon={<PlaceholderIcon />}
           type="switch"
-          switchValue={false}
-          onSwitchValueChange={() => {
-            // TODO: re-enable when ready.
+          switchValue={notificationsPreference === 'enabled'}
+          onSwitchValueChange={isEnabled => {
+            if (isEnabled) {
+              changeNotificationsPreference('enabled');
```

Comment:
Can this logic be encapsulated better so the view doesn't do so much deciding of what happens? Eg a hook that does all this?